### ISSUE TITLE
mongodb "The 'cursor' option is required"

### ISSUE THREAD

**Issue by grooverdan** [MeasureOSS/Measure#99]
MongoDB server version: 3.6.5
Got the same error for quite a few widgets. This was once the RabbitMQ queue from ghcrawler was empty.

From measure @ 791cfbb02c473f4b6826a2ce5809708ff404c711
<pre>
$ node makedash.js
....
  One of the dashboard widgets ("issuesCreatedLine", running 
  against the "null" root had a problem, so I've skipped over it. 
  This is really an internal error, and should be reported. 
  (The error is described like this, which will help in the report: 
  The 'cursor' option is required, except for aggregate with the 
  explain argument, node_modules/mongodb-core/lib/error.js:31) 

...
  One of the dashboard widgets ("usersWithoutOrg", running against 
  the "null" root had a problem, so I've skipped over it. This is 
  really an internal error, and should be reported. 
  (The error is described like this, which will help in the report: 
  The 'cursor' option is required, except for aggregate with the 
  explain argument, node_modules/mongodb-core/lib/error.js:31) 
</pre>

**Comment by stuartlangridge** [comment]
Thank you for the pointer; I'll look into this and see what might be causing it (although the root being called "null" shouldn't happen, and might have something to do with the problem.)

**Comment by grooverdan** [comment]
The root being "null" was odd for me by I was a loss as to what root was. This is my first nodejs app so was at a loss how to debug it or what further info might be useful.

**Comment by grooverdan** [comment]
found in the deadletters of the crawler 'No token with matching traits' for teams, collaborators and events which isn't unexpected as I don't have access on the repository.

**Comment by stuartlangridge** [comment]
I'm a bit baffled by this, because I can't for the life of me make it happen. I do note though that you're on MongoDB 3.6 and I'm on 3.4. Are you using the "docker" setup for ghcrawler? It's possible that something's changed in 3.6 and queries now require a `cursor` option; I can update things to do that, but it's hard to know whether it'll be fixed without me being able to replicate it.

**Comment by grooverdan** [comment]
Yes I'm using the docker ghcrawler which is using the latest tag (https://github.com/Microsoft/ghcrawler/blob/develop/docker/common-services.yml#L28). The SO article lists 3.4.2 in the question so that might not be the full story.

**Comment by grooverdan** [comment]
The errors when way when I did a mongodump and restored to MongoDB v3.4.16
now have site up http://measure-mariadb.ozlabs.org/organizations.html

**Comment by stuartlangridge** [comment]
That's useful info! So there may be an issue with Mongo 3.6....

**Comment by danisyellis** [comment]
I'm getting this same error for a number of my widgets. I'm also using latest mongo, which is currently 4.0.1, with the docker setup. Let me know if you'd like any more information.

**Comment by jeremy-lq** [comment]
Pinning Mongo to 3.4 will work in the interim, until we're able to update support for more recent version.

**Comment by stuartlangridge** [comment]
I am 73.8% sure that this is fixed by https://github.com/MeasureOSS/Measure/commit/6972afd2dd06f0ea41a0b535c0276ba23a267175 now that I've been able to replicate it with Mongo 3.6.
